Libplanet.Action/State/AccountDiff.cs

```diff
@@ -0,0 +1,351 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Security.Cryptography;
+using Bencodex.Types;
+using Libplanet.Common;
+using Libplanet.Crypto;
+using Libplanet.Store.Trie;
+using Libplanet.Types.Assets;
+using Libplanet.Types.Consensus;
+
+namespace Libplanet.Action.State
+{
+    /// <summary>
+    /// Represents a difference between two <see cref="IAccountState"/>s.
+    /// This is a partial interpretation of a raw difference obtained by <see cref="ITrie.Diff"/>
+    /// from <see cref="IAccountState"/>'s perspective.  Keep in mind of the following properties:
+    /// <list type="bullet">
+    ///     <item><description>
+    ///         Any <see lagnword="null"/> value, which is equivalent to non-existent value in
+    ///         the underlying storage, in the source is <em>ignored</em>.  That is, even if
+    ///         the value in the target and the value in the source are different while
+    ///         the value in the source is <see langword="null"/>, this will not be
+    ///         part of the resulting <see cref="AccountDiff"/>.
+    ///     </description></item>
+    ///     <item><description>
+    ///         Any <see langword="null"/> value, again, which is equivalent to non-existent value
+    ///         in the underlying storage, in the target for <see cref="FungibleAssetValue"/>
+    ///         and <see cref="ValidatorSet"/> is interpreted accordingly.  That is,
+    ///         0 amount of <see cref="FungibleAssetValue"/> and empty <see cref="ValidatorSet"/>
+    ///         are used.  This is in accordance with how <see cref="IAccountState.GetBalance"/>
+    ///         and <see cref="IAccountState.GetValidatorSet"/> would behave.
+    ///     </description></item>
+    ///     <item><description>
+    ///         Due to the reason mentioned directly above, the size of <see cref="AccountDiff"/>
+    ///         derived from <see cref="ITrie.Diff"/> may not be the same.  Moreover,
+    ///         an <see cref="AccountDiff"/> being empty <em>does not guarantee</em>
+    ///         that the data are the same as <see cref="IAccountState"/> is not capable of
+    ///         distinguishing between <see langword="null"/> and 0 <see cref="FungibleAssetValue"/>
+    ///         and so on and so forth.
+    ///     </description></item>
+    ///     <item><description>
+    ///         As <see cref="Currency"/> information is in the domain of an application using
+    ///         this library, only the hash of a <see cref="Currency"/> is directly stored in
+    ///         the underlying storage.  As such, each <see cref="Currency"/> and
+    ///         <see cref="FungibleAssetValue"/> are handled as raw values, that is, as hash and
+    ///         <see cref="Integer"/>, for an <see cref="AccountDiff"/>.
+    ///     </description></item>
+    /// </list>
+    /// </summary>
+    public class AccountDiff
```

Comment:
We might be notice this class on CHANGES.md...

Reply:
Done. 😗